File: src/BTrade3/app/src/main/java/com/elsasa/btrade3/model/CheckOutMode.kt
package com.elsasa.btrade3.model

/**
 * Indicates how a visit was closed. Stored as string in Room for forward compatibility.
 */
object CheckOutMode {
    const val MANUAL = "MANUAL"
    const val AUTO = "AUTO"

    fun isValid(value: String): Boolean = value == MANUAL || value == AUTO
}

File: src/j05-btr-distrib/btr.domain/SalesContext/CheckInFeature/CheckOutMode.cs
namespace btr.domain.SalesContext.CheckInFeature
{
    /// <summary>
    /// Indicates how a visit was closed.
    /// </summary>
    public static class CheckOutMode
    {
        public const string Manual = "MANUAL";
        public const string Auto = "AUTO";

        public static bool IsValid(string value)
        {
            return value == Manual || value == Auto;
        }
    }
}
